framework/PostProcessors/ValueDuration.py

```diff
@@ -0,0 +1,168 @@
+
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.numBins = None # integer number of bins to use in creating the duration curve. TODO default?
+    self.dynamic = True
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = self.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    for child in paramInput.subparts:
+      tag = child.getName()
+      if tag == 'target':
+        self.targets = set(child.value)
+      elif tag == 'bins':
+        self.numBins = child.value
+
+  def initialize(self, runInfo, inputs, initDict):
+    """
+      Method to initialize the pp.
+      In here the working dir is grepped.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def inputToInternal(self, currentInp):
+    """
+      Method to convert an input object into the internal format that is
+      understandable by this pp.
+      In this case, we only want data objects!
+      @ In, currentInp, object, an object that needs to be converted
+      @ Out, inputDict, dict, dictionary of the converted data
+    """
+    if len(currentInp) > 1:
+      self.raiseAnError(IOError, 'Expected 1 input HistorySet, but received {} inputs!'.format(len(currentInp)))
+    currentInp = currentInp[0]
+    if currentInp.type not in ['HistorySet']:
+      self.raiseAnError(IOError, 'ValueDuration postprocessor "{}" requires a HistorySet input! Got "{}".'
+                                 .format(self.name, currentInput.type))
+    return currentInp
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action.
+      @ In, inputIn, object, object contained the data to process. (inputToInternal output)
+      @ Out, realizations, list, list of realizations obtained
+    """
+    inData = self.inputToInternal(inputIn)
+    realizations = []
+    # new load duration curve for each sample
+    for s in range(len(inData)):
+      # obtain relevant sample
+      sample = inData.realization(index=s)
+      # solutions for each realization are stored in this dict
+      rlz = {}
+      # separate curve for each target
+      for target in self.targets:
```

Comment:
using two loops can be slow when the input large datasets

Reply:
It can.  In the event this becomes a burden, we can write a function to "apply" with, which may not be much faster, but could possibly speed up and be applied with Dask better.